src/RA_Dlg_AchEditor.cpp

```diff
@@ -1697,9 +1680,43 @@ INT_PTR Dlg_AchievementEditor::AchievementEditorProc(HWND hDlg, UINT uMsg, WPARA
                                 if (pConfiguration.IsFeatureEnabled(ra::services::Feature::PreferDecimal))
                                     nBase = 10;
                             }
-
-                            const auto nVal = std::strtoul(sData.c_str(), nullptr, nBase);
-                            rCond.CompTarget().SetValue(nVal);
+                            
+                            auto lVal = 0UL;
+                            auto lineErrNumber = 0;
+                            try
+                            {
+                                lineErrNumber = __LINE__ + 1;
+                                lVal = std::stoul(sData.c_str(), nullptr, nBase);
+                            } catch (const std::invalid_argument& e)
+                            {
+                                RA_LOG_ERR("\nFunction: %s\nLine: %d\nMessage: %s\n\n", __FUNCTION__, lineErrNumber,
+                                           e.what());
+                                const auto bIsDecimal = Button_GetCheck(::GetDlgItem(hDlg, IDC_RA_CHK_SHOW_DECIMALS));
+                                if (bIsDecimal)
+                                {
+                                    ra::ui::viewmodels::MessageBoxViewModel::ShowWarningMessage(L"Invalid Input",
+                                        L"Only values that can be represented as decimal are allowed while the 'show "
+                                        L"decimal values' checkbox is "
+                                        L"checked.\n\nPlease try again!");
+                                }
+                                else
+                                {
+                                    ra::ui::viewmodels::MessageBoxViewModel::ShowWarningMessage(L"Invalid Input",
+                                        L"Only values that can be represented as hexadecimal are allowed while the "
+                                        L"'show decimal values' checkbox is unchecked.\n\nPlease try again!");
+                                }
+                            } catch (const std::out_of_range& e)
+                            {
+                                RA_LOG_ERR("Function: %s\nLine: %d\nMessage: %s", __FUNCTION__, lineErrNumber,
+                                           e.what());
+                                constexpr auto maxUlong = std::numeric_limits<unsigned long>::max();
+                                ra::ui::viewmodels::MessageBoxViewModel::ShowWarningMessage(
+                                    L"Too Large",
+                                    ra::StringPrintf(L"Value supplied is negative or too large!\nIt must be "
+                                                     L"non-negative and less than %lu (decimal) or 0x%X (hexadecimal)",
+                                                     maxUlong, maxUlong));
+                            }
+                            rCond.CompTarget().SetValue(lVal);
```

Comment:
Everything works as intended here. Type in the wrong input and the correct warning will be displayed but will be logged as an error. Say if you typed `9999999999999999999999999999`, an out_of_range will happen.

Type "bad" or w/e when the decimal button is checked, it'll tell you you can only use decimal values. Any character that can't be represented as a hexadecimal will get the other warning.

Had to change `AppendPrintf `a bit to write out a non-truncated hexadecimal number.